### DISCUSSION TITLE
Can HEXL implement modulus add/ modulus subtract/ modulus mul without using divide in hardware?

### DISCUSSION THREAD

**Discussion by sjain70** [intel/hexl#132] [Q&A] [answered]
* does hexl implement modulo operations on CPU ALU also? 
* Can hexl implement modulo operations without divide operation? My understanding is that CPU implements modulus as a division returning both quotient and remainder in different registers.

**Comment by joserochh** [comment, marked as answer]
Sorry for the late response. I was not being notified of this conversations. Yes, HEXL can perform those operations without divisions by using Barrett Reduction.
